src/core/reference/include/ngraph/runtime/reference/scatter_elements_update.hpp

```diff
@@ -165,6 +162,25 @@ size_t normalize_index(const T idx, const size_t dim_value) {
     }
 }
 
+template <typename T>
+struct RoundingDirectionGuard {
+    RoundingDirectionGuard() {
+        if (std::is_integral<T>::value) {
+            m_original_mode = std::fegetround();
+            std::fesetround(FE_DOWNWARD);
```

Comment:
is it thread safe?

Reply:
"It is thread-local. Each thread inherits the initial state of its floating-point environment from the parent thread."
Source: https://en.cppreference.com/w/cpp/numeric/fenv